### PR TITLE
Remove calls of  c10::guts::conjunction,c10::guts::disjunction,c10::guts::negation

### PR DESCRIPTION
**Pull request by cyyever**
cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: c10/util/C++17.h [hunk 1]
```diff
@@ -73,48 +73,6 @@ make_unique_base(Args&&... args) {
   return std::unique_ptr<Base>(new Child(std::forward<Args>(args)...));
 }
 
-#if defined(__cpp_lib_logical_traits) && !(defined(_MSC_VER) && _MSC_VER < 1920)
-
-template <class... B>
```
Comment (Skylion007) [inline c10/util/C++17.h:78]: for backwards compatability shouldn't this if_def becomes unconditional alias here (for internal meta code and Python extensions that might uses aliases features)?

Comment (cyyever) [reply c10/util/C++17.h:78]: It is actually safer to keep the templates.

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**